tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedEventMatcher.cs

```diff
@@ -0,0 +1,167 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Events;
+using Xunit.Sdk;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public class UnifiedEventMatcher
+    {
+        private UnifiedValueMatcher _valueMatcher;
+
+        public UnifiedEventMatcher(UnifiedValueMatcher valueMatcher)
+        {
+            _valueMatcher = valueMatcher;
+        }
+
+        public void AssertEventsMatch(List<object> actualEvents, BsonArray expectedEventsDocuments)
+        {
+            try
+            {
+                AssertEvents(actualEvents, expectedEventsDocuments);
+            }
+            catch (XunitException exception)
+            {
+                throw new AssertionException(
+                    userMessage: GetAssertionErrorMessage(actualEvents, expectedEventsDocuments),
+                    innerException: exception);
+            }
+        }
+
+        // private methods
+        private void AssertEvents(List<object> actualEvents, BsonArray expectedEventsDocuments)
+        {
+            actualEvents.Should().HaveSameCount(expectedEventsDocuments);
+
+            for (int i = 0; i < actualEvents.Count; i++)
+            {
+                var actualEvent = actualEvents[i];
+                var expectedEventDocument = expectedEventsDocuments[i].AsBsonDocument;
+                if (expectedEventDocument.ElementCount != 1)
+                {
+                    throw new FormatException("Expected event document model must contain a single element");
+                }
+                var expectedEventType = expectedEventDocument.GetElement(0).Name;
+                var expectedEventValue = expectedEventDocument[0].AsBsonDocument;
+
+                switch (actualEvent)
+                {
+                    case CommandStartedEvent commandStartedEvent:
+                        expectedEventType.Should().Be("commandStartedEvent");
```

Comment:
This is wrong. In FluentAssertions the actual value always comes first.

Replace 66-69 with:

    switch (expectedEventType)
    {
        case "commandStartedEvent":
            var commandStartedEvent = actualEvent.Should().BeOfType<CommandStartedEvent>().Subject;

The rest of the `case` code can remain the same.